addons/test_mail/tests/test_mail_thread_internals.py

```diff
@@ -454,6 +454,24 @@ def test_unlink_notification_message(self):
 class TestNoThread(MailCommon, TestRecipients):
     """ Specific tests for cross models thread features """
 
+    @users('employee')
+    def test_mail_sending_on_non_thread_model(self):
+        """ This test simulates scenarios where a required method called `_process_attachments_for_post` is missing,
+        in such case composer should fallback to the method implementation in mail.thread. """
+        template = self.env['mail.template'].create({
+            'name': 'TestTemplate',
+            'model_id': self.env['ir.model']._get_id('mail.test.nothread'),
+        })
+        mail_compose_message = self.env['mail.compose.message'].create({
```

Comment:
actually you can give res_ids and then do composer.action_send_mail so that you don't have to call internals :)